src/main/java/sirius/biz/storage/layer3/MutableVirtualFile.java

```diff
@@ -40,15 +42,44 @@ protected MutableVirtualFile() {
     }
 
     /**
-     * Creates a new file with the given name.
+     * Use the static factory methods to obtain a new instance.
      *
-     * @param parent the parent of the file
-     * @param name   the name of the file
+     * @see #create(VirtualFile, String)
+     * @see #safeCreate(VirtualFile, String)
      */
     protected MutableVirtualFile(@Nonnull VirtualFile parent, @Nonnull String name) {
         super(parent, name);
     }
 
+    /**
+     * Creates a new file with the given name in the given directory.
+     *
+     * @param parent the parent of the file
+     * @param name   the name of the file
+     * @return a new instance of this class
+     * @throws IllegalArgumentException if the given name is empty or contains '/'
+     */
+    public static MutableVirtualFile create(@Nonnull VirtualFile parent, @Nonnull String name) {
+        return new MutableVirtualFile(parent, name);
+    }
+
+    /**
+     * Creates a new file with the given name in the given directory.
+     * <p>
+     * Replaces any '/' chars in the given String and if the given String is empty, <tt>null</tt> is returned.
+     *
+     * @param parent the parent of the file
+     * @param name   the name of the file
+     * @return a new instance of this class
+     */
+    @Nullable
+    public static MutableVirtualFile safeCreate(@Nonnull VirtualFile parent, @Nullable String name) {
+        if (Strings.isEmpty(name)) {
+            return null;
+        }
+        return new MutableVirtualFile(parent, name.replace('/', '_'));
```

Comment:
Do we need to make sure these file names are also Windows-compatible? If so, we should include a few more characters to the list.

- Linux/Unix: / (forward slash)
- Windows: < (less than) > (greater than) : (colon - sometimes works, but is actually NTFS Alternate Data Streams) " (double quote) / (forward slash) \ (backslash) | (vertical bar or pipe) ? ( question mark) * (asterisk)

Reply:
I think we only have to regard the SFTP restrictions for filenames

Reply:
Maybe blocking some more restricted characters would be a sane approach and wouldn't hurt too much(?)

https://en.wikipedia.org/wiki/Filename has a good table - and everythign except . and " " seems dangerous. Maybe we provide three methods one that crashes, one that replaces and one that accepts everything (while trusting the caller) to properly check and format the string(?)